File: drs/src/core/com/db4o/drs/foundation/Signatures.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.drs.foundation;

import java.util.*;

public class Signatures {
	
	private final Map<Signature, Long> _loidBySignature = new HashMap<Signature, Long>();

	private final Map<Long, Signature> _signatureByLoid = new HashMap<Long, Signature>();

	public void add(Signature signature, long signatureLoid) {
		_loidBySignature.put(signature, signatureLoid);
		_signatureByLoid.put(signatureLoid, signature);
	}
	
	public Long loidForSignature(Signature signature){
		return _loidBySignature.get(signature);
	}
	
	public Signature signatureForLoid(long signatureLoid){
		return _signatureByLoid.get(signatureLoid);
	}

}

File: db4o.net/drs/Db4objects.Drs/Db4objects.Drs/Foundation/Signatures.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System.Collections.Generic;
using Db4objects.Drs.Foundation;

namespace Db4objects.Drs.Foundation
{
	public class Signatures
	{
		private readonly IDictionary<Signature, long> _loidBySignature = new Dictionary<Signature
			, long>();

		private readonly IDictionary<long, Signature> _signatureByLoid = new Dictionary<long
			, Signature>();

		public virtual void Add(Signature signature, long signatureLoid)
		{
			_loidBySignature[signature] = signatureLoid;
			_signatureByLoid[signatureLoid] = signature;
		}

		public virtual long LoidForSignature(Signature signature)
		{
			return _loidBySignature[signature];
		}

		public virtual Signature SignatureForLoid(long signatureLoid)
		{
			return _signatureByLoid[signatureLoid];
		}
	}
}
